pkg/controller/storagecluster/reconcile.go

```diff
@@ -530,6 +530,11 @@ func determinePlacementRack(nodes *corev1.NodeList, node corev1.Node, minRacks i
 	}
 
 	sort.Strings(rackList)
+	if len(rackList) == 0 {
```

Comment:
instead of `len(rackList) == 0`, I recommend `if len(tartgetAZ) == 0`, then:

```
if len(targetA) > 0 {
    # old method for pupulating rackList
} else {
    # your new method
}